nncf/torch/graph/node_utils.py

```diff
@@ -0,0 +1,44 @@
+"""
+ Copyright (c) 2023 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from torch import Tensor
+from nncf.common.graph.graph import NNCFNode
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.graph.operator_metatypes import OPERATORS_WITH_BIAS_METATYPES
+
+
+def is_node_with_bias(node: NNCFNode, model: NNCFNetwork) -> bool:
+    """
+    Checks if the node has a bias or not.
+
+    :param node: The node to check.
+    :param model: The model that contains this operation.
+    :return: Return `True` if `node` corresponds to the operation
+        with bias (bias is added to the output tensor of that operation),
+        `False` otherwise.
+    """
+    node_module = model.get_containing_module(node.node_name)
+    return node.metatype in OPERATORS_WITH_BIAS_METATYPES and node_module.bias is not None
+
+
+def get_bias_value(node: NNCFNode, model: NNCFNetwork) -> Tensor:
+    """
+    Returns the bias tensor for the biased node.
+
+    :param node: The node that corresponds to the operation with bias.
+    :param model: The model that contains this operation.
+    :return: The bias value that is applied to the output tensor of the node's operation.
+    """
+    node_module = model.get_containing_module(node.node_name)
+    if node_module.bias is None:
+        return None
```

Comment:
So, what would happen in case of `None`? How this condition might work in case if we check `is_node_with_bias` previously?

Reply:
> So, what would happen in case of None? 

For FBC, it checks in function `is_node_with_bias` before using `get_bias_value`, that guarantied that bias will be `not None`.


>  How this condition might work in case if we check is_node_with_bias previously?

What do you mean? 
How is a check to `None` in function `get_bias_value` connected with `is_node_with_bias`?

Reply:
> Ok. Then, this code is useless. Am I right?

No, it separated functions. `get_bias_value` can be reused without `is_node_with_bias`. 
In your logic `get_bias_value` should raise exception `NoneType is not data attribute` if node without bias?

Which is problem here?

Reply:
The problem with the returned value is that can be `None`. Is that the `None` value catches somewhere further in the code? No. This is the problem.

Reply:
> Is that the None value catches somewhere further in the code? 

1. As i said for FBC, it checks in function `is_node_with_bias` before using get_bias_value, that guarantied that bias will be not None.
2. If `get_bias_value` without `is_node_with_bias`, yes need add check to None.
3. Do you suggest add `if bias is None:` after `get_bias_value`? It will be double check. Are your really sure that it need?

Reply:
> As i said for FBC, it checks in function is_node_with_bias before using get_bias_value, that guarantied that bias will be not None.

It means that `if node_module.bias is None:` is useless, isn't it? 

> If get_bias_value without is_node_with_bias, yes need add check to None.

So, add it, please.

> Do you suggest add if bias is None: after get_bias_value? It will be double check. Are your really sure that it need?

No. Just do not add unnecessary code. Why unnecessary? See point 1.

Reply:
> It means that if node_module.bias is None: is useless, isn't it?

No, `get_bias_value` function is not FBC specific private function, it can be reused separately from `is_node_with_bias`. And function should not raise Exception.


So, you says that check `if node_module.bias is None:` in `get_bias_value` is useless, because it checks before `is_node_with_bias` and it will be always false. In same time you want to add `if bias is None:` that will be always False for same reason.
Do I correctly understand you?

>So, add it, please.
Deal, I will rework this moment.

Reply:
Remove `node_and_bias_value` generator, looks so large. Moved all checks in loop `for node in nncf_graph.get_all_nodes():` with check None.